src/communication/CommandLine.cpp

```diff
@@ -204,14 +193,12 @@ void CommandLine::sliceNext()
                     force_read_nondefault = false;
                 }
 #ifdef __EMSCRIPTEN__
-                else if (argument.find("--progress") == 0)
+                else if (argument.starts_with("--progress_cb") || argument.starts_with("--slice_info_cb"))
                 {
-                    // Store progress handler name
                     argument_index++;
                     argument = arguments_[argument_index];
-                    progressHandler = argument;
                 }
-#endif
+#endif // __EMSCRIPTEN__
```

Comment:
Shouldn't this part be moved to the dedicated class ?

Reply:
_This is me, trying very hard not to rewrite our custom command line parsing logic_. I normally agree with you. But solving this properly would be more work, this was the quickest fix.

I am tempted to use cxxopts https://github.com/jarro2783/cxxopts an open a community PR one of these days.

Reply:
Let me know how strong you feel about letting this through

Reply:
Well it just feels sad that you took a lot of effort to split the Emscripten part out of CommandLine, and yet you still have this dirty part remaining there :smiley:
You could solve this easily by having a virtual method to parse extra arguments, that would return false by default and would parse those in the `EmscriptenCommunication` overridden definition.

Reply:
Yes, it feels dirty, Let's see if I can manage to find some time to make the changes for a cleaner solution